src/MetadataManager/EventManager.ts

```diff
@@ -58,34 +61,63 @@ export class MetadataEventManager {
-          let path='a.log';
-          console.log(1);
-          let hash=await Metadata.contentHash(path);
-          console.log(hash);
-          let content=await Metadata.getMetadata(hash);
-          console.log(content);
-          content['b.log']="Test";
-          await Metadata.setMetadata(hash, content);
-        }
+        provider.refresh('Create'); // test code
+        console.log('onDidCreate  '+uri.fsPath);
+        MetadataEventManager.createUri=uri;
+        //if dir
+        // [case 5] > (1) call search > listup files > while [case4]
+        //else files
+        // validcheck(endswith)
+        // pathToHash search [case 1] > (1) call contentHash (2) change pathToHash (3) deactivate hash from pathToHash (4) insert hash from contentHash
+        // [case 4] > (1) call contentHash (2) getMetadata (3) compare to path(activate?) > Yes(ignore), No(Setup)
+        let temp=await vscode.workspace.findFiles('a.log/**');
+        console.log(temp);
+        timerId=setTimeout(()=>{MetadataEventManager.createUri=undefined; console.log('test  '+ MetadataEventManager.createUri);},0);
+        ////// case 1. [File] Contents change event (refer to pathToHash)
+        ////// case 2(ignore). [File] Move contents > Processing in Delete
+        ////// case 3(ignore). [Dir]  Move contents > Processing in Delete (reconginition Dir when Dir+File moved)
+        // case 4. [File] Copy many files
+        // case 5. [Dir]  Copy with files > Serch all the file in the Dir
+        // *if already exist, ignore the creation events.
+        // *always new path.
+        // *
+        // case 4. [File] Generate Product from ONE (processing like case 1 or ignore)
       }),
       provider.fileWatcher.onDidChange(uri => {
         console.log(uri); provider.refresh('Change'); // test code
         // case 1. Contents change event only > command event
       }),
-      provider.fileWatcher.onDidDelete(uri => { // To Semi Jeong
-        console.log(uri); provider.refresh('Delete'); // test code
-        // case 1. Delete file (Metadata deactivate or ignore)  > command event
+      provider.fileWatcher.onDidDelete(async uri => { // To Semi Jeong
+        console.log('onDidDelete::', uri); provider.refresh('Delete'); // test code
+        // TODO: pathToHash update
+        // 만약 필요하다면 pathToHash에도 folder용 update, file용 update 만들어서 따로 처리 (2번 돌아서 비효율적)
+        // 아니면 그냥 현재 disableMetadata, moveMetadata(file용 함수)에서만 처리하기
+        // file: 일반 삭제 > 그냥 지우기, 이동/rename > path 이름 변경
+        const path = uri.path;
+        if (MetadataEventManager.createUri) {
+          const newUri = MetadataEventManager.createUri.path;
+          // The file/folder is moved/renamed
+          if (fs.statSync(newUri).isDirectory()) {
+            // case 4. [Dir]+Path       | move > search (delete & new)
+            Metadata.moveMetadataUnderFolder(path, newUri);
+          } else if (provider.isValidFile(path)) { // FIXME: Do we have to check isValidFile for newUri too?
```

Comment:
I'm not sure that we need to lose control for a renamed file which was controlled our system. (ex. renamed from 'a.log' to 'a.txt')
Do we have to check isValidFile for newUri too?